src/Resources/config/services.xml

```diff
@@ -13,6 +13,9 @@
             <argument type="service" id="service_container" />
             <argument /> <!-- argument added dynamically -->
         </service>
+	
+	<!-- Add service alias for autowiring -->
+	<service id="KnpU\OAuth2ClientBundle\Client\ClientRegistry" alias="knpu.oauth2.registry" public="false" />
```

Comment:
Shouldn't we rename `knpu.oauth2.registry` to `KnpU\OAuth2ClientBundle\Client\ClientRegistry` and do the reverse?
```xml
<service id="knpu.oauth2.registry" alias="KnpU\OAuth2ClientBundle\Client\ClientRegistry" public="false" />
```